### PR TITLE
showing downloaded patient list from DB if sync Api fails in between

### LINKED ISSUE #1134: Application's first launch, Downloading patient (resources) failing with SocketTimeoutException
**Describe the bug**
Install and launch app for first time, downloading fails with SocketTimeoutException after around 12min

**Component**
Core library, reference app

**To Reproduce**
1. Install Demo Application on Android phone
2. Launch App
3. Wait for automatically download patients(resources)
4. See behaviour : App does not show patient list
5. but on app relaunch it shows the patient list

**Expected behavior**
Patient list should be shown to the user on application launch, if list is long it should download and show items on scroll.

**Screenshots**
OnFirstLaunchEmptyPatientList

**Smartphone (please complete the following information):**
 - Device: Nokia 3.4
 - Android version: 11
 - Build number: 00WW_2_410_SP01

**Additional context**
For smaller data app works fine. Issue exists only for large data

### PR DESCRIPTION
**Pull request by PallaviGanorkar**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1134

**Description**
1. Added progress bar for sync operation at first app launch, So the screen won't be Idle while downloading resources
2. If the some resources are downloaded and the API fails in between, Here shown list from DB wherever patients are downloaded.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by jingtang10** [comment]
great pr - thanks @PallaviGanorkar!

can you attach a video?

**Review by jingtang10 — CHANGES_REQUESTED** [review]
@aditya-07 suggested having a top banner that shows the status fo the sync. I think that makes this implementation easier. bcause you don't need to use shared preferences any more. just always listen to the state and update the top banner.

**Comment by jingtang10** [comment]
@shelaghm just so you are aware.

**Comment by PallaviGanorkar** [comment]
video attached with suggested UI change by @shelaghm 

https://user-images.githubusercontent.com/13957417/171858835-a5c18a1e-115d-4ebb-9fca-a730f29f8c37.mp4

**Comment by PallaviGanorkar** [comment]
sync status background color updated as per UI

sync_status_background_color

**Review by jingtang10 — CHANGES_REQUESTED** [review]
@PallaviGanorkar can you also remove the toast in the same PR?

File: demo/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -66,4 +66,6 @@
   <string name="close">Close</string>
   <string name="sync_menu">Sync</string>
   <string name="last_sync">Last sync</string>
+  <string name="is_first_launch">isFirstLaunch</string>
```
[the remark this answers is not in the document]
Comment (PallaviGanorkar) [reply demo/src/main/res/values/strings.xml:69]: Done

File: demo/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -66,4 +66,6 @@
...
+  <string name="syncing">SYNCING</string>
```
[the remark this answers is not in the document]
Comment (PallaviGanorkar) [reply demo/src/main/res/values/strings.xml:70]: Done

File: demo/src/main/res/layout/sync_status_layout.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<LinearLayout
```
[the remark this answers is not in the document]
Comment (PallaviGanorkar) [reply demo/src/main/res/layout/sync_status_layout.xml:2]: i am using  android:drawableLeft in TextView, this is case  drawable will be displayed on extreme left of TextView with match parent width and 'Syncing' text will appear at center. So, to display drawable and text at center have to use layout

File: demo/src/main/java/com/google/android/fhir/demo/PatientListFragment.kt [hunk 4]
```diff
@@ -47,6 +48,7 @@ class PatientListFragment : Fragment() {
   private lateinit var fhirEngine: FhirEngine
   private lateinit var patientListViewModel: PatientListViewModel
   private lateinit var searchView: SearchView
+  private lateinit var tvSyncingStatus: LinearLayout
```
[the remark this answers is not in the document]
Comment (PallaviGanorkar) [reply demo/src/main/java/com/google/android/fhir/demo/PatientListFragment.kt:51]: Done

File: demo/src/main/java/com/google/android/fhir/demo/PatientListFragment.kt [hunk 5]
```diff
@@ -143,8 +146,17 @@ class PatientListFragment : Fragment() {
...
         Timber.d("onViewCreated: pollState Got status $it")
+        if (it is State.Started || it is State.InProgress) {
+          if (tvSyncingStatus.visibility != View.VISIBLE) {
+            tvSyncingStatus.visibility = View.VISIBLE
+          }
+        }
+
         // After the sync is successful, update the patients list on the page.
-        if (it is State.Finished) {
+        if (it is State.Finished || it is State.Failed) {
```
[the remark this answers is not in the document]
Comment (PallaviGanorkar) [reply demo/src/main/java/com/google/android/fhir/demo/PatientListFragment.kt:165]: Done these in this PR itself. Please review

**Pull request merged**